coniql/caplugin.py

```diff
@@ -134,18 +168,38 @@ async def put_channels(
     async def subscribe_channel(
         self, pv: str, config: ChannelConfig
     ) -> AsyncIterator[Channel]:
-        q: asyncio.Queue[AugmentedValue] = asyncio.Queue()
-        info, meta_value = await asyncio.gather(
-            cainfo(pv), caget(pv, format=FORMAT_CTRL),
+        channel = CAChannel(pv, config)
+        # A queue that contains a monitor update and the keyword with which
+        # the channel's update_value function should be called.
+        q: asyncio.Queue[Tuple[str, AugmentedValue]] = asyncio.Queue()
+        # Monitor PV for value and alarm changes with associated timestamp.
+        value_monitor = camonitor(
+            pv, lambda v: q.put(("time_value", v)), format=FORMAT_TIME,
+        )
+        # Monitor PV only for property changes. For EPICS < 3.15 this monitor
+        # will update once on connection but will not subsequently be triggered.
+        # https://github.com/dls-controls/coniql/issues/22#issuecomment-863899258
+        meta_monitor = camonitor(
+            pv,
+            lambda v: q.put(("meta_value", v)),
+            events=DBE_PROPERTY,
+            format=FORMAT_CTRL,
         )
-        m = camonitor(pv, q.put, format=FORMAT_TIME)
         try:
-            # Hold last channel for squashing identical alarms
-            last_channel = None
+            target, first_value = await q.get()
+            yield channel.update_value(**{target: first_value})
+            # A specific request required for whether the channel is writeable.
+            # This will not be updated, so wait until a callback is received
+            # before making the request when the channel is likely be connected.
+            try:
+                info = await cainfo(pv)
+                yield channel.update_value(writeable=info.write)
+            except CANothing:
+                pass  # Allow subscriptions to continue
```

Comment:
When would we trigger this case?

Reply:
If the channel disconnected in between sending an update and the cainfo request. Unlikely but possible?

Reply:
I guess so...